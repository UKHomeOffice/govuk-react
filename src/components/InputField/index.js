// https://github.com/alphagov/govuk-frontend/blob/master/src/components/input/_input.scss
// https://github.com/alphagov/govuk_elements/blob/master/assets/sass/elements/_forms.scss

import React from "react";
import PropTypes from "prop-types";

import Label from "../Label/index";
import LabelText from "../LabelText/index";
import ErrorText from "../ErrorText/index";
import HintText from "../HintText/index";
import Input from "../Input/index";

const InputField = ({ meta, children, hint, input }) => (
  <Label error={meta.touched && meta.error}>
    <LabelText>{children}</LabelText>
    {hint && <HintText>{hint}</HintText>}
    {meta.touched && meta.error && <ErrorText>{meta.error}</ErrorText>}
    <Input {...input} />
  </Label>
);

InputField.defaultProps = {
  hint: null,
  input: {},
  meta: {}
};

InputField.propTypes = {
  hint: PropTypes.string,
  input: PropTypes.shape({
    name: PropTypes.string,
    onBlur: PropTypes.func,
    onChange: PropTypes.func,
    onFocus: PropTypes.func,
    value: PropTypes.any
  }),
  meta: PropTypes.shape({
    active: PropTypes.bool,
    dirty: PropTypes.bool,
    dirtySinceLastSubmit: PropTypes.bool,
    error: PropTypes.any,
    initial: PropTypes.bool,
    invalid: PropTypes.bool,
    pristine: PropTypes.bool,
    submitError: PropTypes.any,
    submitFailed: PropTypes.bool,
    submitSucceeded: PropTypes.bool,
    touched: PropTypes.bool,
    valid: PropTypes.bool,
    visited: PropTypes.bool
  }),
  children: PropTypes.node.isRequired
};

export default InputField;
