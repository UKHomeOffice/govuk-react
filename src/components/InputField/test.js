import React from "react";
import ReactDOM from "react-dom";
import { shallow, mount, render } from "enzyme";

import InputField from "./";

describe("InputField", () => {
  const example = "example";
  const wrapper = <InputField>{example}</InputField>;

  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(wrapper, div);
  });

  it('should render an input type="text"', () => {
    const output = shallow(wrapper);
    expect(output.find('input[type="text"]')).toBeTruthy();
  });

  it("should render a label", () => {
    const output = shallow(wrapper);
    expect(output.find("label")).toBeTruthy();
  });

  it("matches snapshot", () => {
    expect(shallow(wrapper)).toMatchSnapshot(`enzyme.shallow`);
    expect(mount(wrapper)).toMatchSnapshot(`enzyme.mount`);
    expect(render(wrapper)).toMatchSnapshot(`enzyme.render`);
  });
});
