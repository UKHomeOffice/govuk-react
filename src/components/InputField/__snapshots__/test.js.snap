// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputField matches snapshot: enzyme.mount 1`] = `
<InputField
  hint={null}
  input={Object {}}
  meta={Object {}}
>
  <glamorous(label)>
    <label
      className="css-wd82g1"
    >
      <glamorous(span)>
        <span
          className="css-1m1nyko"
        >
          example
        </span>
      </glamorous(span)>
      <glamorous(input)
        type="text"
      >
        <input
          className="css-opcxs3"
          type="text"
        />
      </glamorous(input)>
    </label>
  </glamorous(label)>
</InputField>
`;

exports[`InputField matches snapshot: enzyme.render 1`] = `
<label
  class="css-wd82g1"
>
  <span
    class="css-1m1nyko"
  >
    example
  </span>
  <input
    class="css-opcxs3"
    type="text"
  />
</label>
`;

exports[`InputField matches snapshot: enzyme.shallow 1`] = `
<glamorous(label)>
  <glamorous(span)>
    example
  </glamorous(span)>
  <glamorous(input)
    type="text"
  />
</glamorous(label)>
`;
